server/kafka/partitioner.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package kafka
+
+import (
+	"github.com/Shopify/sarama"
+	"sync"
+)
+
+// A balancer (or partitioner in Sarama terms) is in-charge of spreading messages across available partitions of a topic.
+// Sarama provides a hash based partitioner which is default in the producer. This implementation distributes messages
+// based on number of bytes each partition has received.
+type leastBytesPartitioner struct {
+	mutex        sync.Mutex
+	byteCounters map[int]uint64
+}
+
+func NewLeastBytesPartitioner(topic string) sarama.Partitioner {
```

Comment:
Somewhere around here we should document that this function implements https://pkg.go.dev/github.com/Shopify/sarama#PartitionerConstructor and that `topic` is not used, but is part of the interface.

Reply:
Done.